### PR TITLE
move idle functionality into room model and out of worker

### PR DESCRIPTION
**Pull request by sisk**

### CODE REVIEW

File: app/models/room.rb [hunk 1]
```diff
@@ -9,4 +9,14 @@ class Room < ApplicationRecord
...
   has_one :current_song, through: :current_record, source: :song
   belongs_to :team
+
+  def idle!
+    just_finished = current_record.present?
+    update!(current_record: nil, playing_until: nil, waiting_songs: false)
+
+    return unless just_finished
+
+    BroadcastNowPlayingWorker.perform_async(self.id)
```
Comment (sisk) [inline app/models/room.rb:19]: I wasn't 100% happy bout leaving these worker calls in here. It may be argued that this refactor is not necessary.

Comment (trumanshuck) [reply app/models/room.rb:19]: You know, I think that the early return around this `just_finished` concept was important to us when the (at the time) `QueueManagemenWorker` was responsible for queuing itself over and over.  The expectation there was that if no one was playing music in a room, it'd hit this "empty" block every one or two seconds, so we wanted to skip enqueuing these broadcasts.

With the event loop thing, we're more or less guaranteed that `just_finished` should always be true.  I like encapsulating the `update!` in this `idle!` method -- perhaps we could remove `just_finished` and pull the broadcasts back out and invoke them just after `idle!`?

Comment (trumanshuck) [reply app/models/room.rb:19]: Oh we early return `idle!` so it's not quite that straightforward, but still perhaps something to think about.

**Comment by sisk** [comment]
Just saw @trumanshuck comments in Campfire. This can be closed if not needed.

**Pull request closed without merging**